### PR TITLE
feat: cssinjs for menu

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       cssinjs for menu    |
| 🇨🇳 Chinese |        menu 组件改为 cssinjs    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by MadCcc** [comment]
image
少了下面的边

File: components/menu/style/index.tsx [hunk 1]
```diff
@@ -1,6 +1,1064 @@
...
-// style dependencies
-// deps-lint-skip: layout
-import '../../tooltip/style';
+/** Component only token. Which will handle additional calculation of alias token */
+export interface ComponentToken {
+  antPrefix: string;
+  componentBackground: string;
+  colorTextSecondary: string;
+  motionDurationMD: string;
+  primaryColor: string;
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/menu/style/index.tsx:13]: 上下都是，重复的都去掉即可。

File: components/menu/style/index.tsx [hunk 2]
```diff
@@ -1,6 +1,1064 @@
...
+  boxShadowColor: string;
+  highlightDangerColor: string;
+  itemActiveDangerBg: string;
+}
+
+interface MenuToken extends FullToken<'Menu'> {}
+
+// =============================== Base ===============================
+const accessibilityFocus: GenerateStyle<MenuToken, CSSObject> = token => ({
+  boxShadow: `0 0 0 2px ${token.colorPrimarySecondary}`,
```
Comment (zombieJ) [inline components/menu/style/index.tsx:40]: `FIXME`

File: components/menu/style/index.tsx [hunk 3]
```diff
@@ -1,6 +1,1064 @@
...
+      [`
+        ${componentCls}-item:hover,
+        ${componentCls}-item-active,
+        ${componentCls}:not(${componentCls}-inline) ${componentCls}-submenu-open,
+        ${componentCls}-submenu-active,
+        ${componentCls}-submenu-title:hover
+      `]: {
+        color: primaryColor,
+        [`${componentCls}`]: {
+          color: '#000',
```
Comment (zombieJ) [inline components/menu/style/index.tsx:109]: `FIXME`

File: components/menu/style/index.tsx [hunk 4]
```diff
@@ -1,6 +1,1064 @@
...
+      },
+
+    // dark theme
+    [`&${componentCls}-dark,${componentCls}-dark ${componentCls}-sub,&${componentCls}-dark ${componentCls}-sub`]:
+      {
+        color: darkColor,
+        background: darkBg,
+        [`${componentCls}-submenu-title ${componentCls}-submenu-arrow`]: {
+          opacity: 0.45,
+          transition: 'all 0.3s',
```
Comment (zombieJ) [inline components/menu/style/index.tsx:136]: token.durationSlow 代替

File: components/menu/style/index.tsx [hunk 5]
```diff
@@ -1,6 +1,1064 @@
...
+      paddingInlineStart: 0, // Override default ul/ol
+      color: colorText,
+      fontSize,
+      lineHeight: 0, // Fix display inline-block gap
+      textAlign: 'left',
+      listStyle: 'none',
+      background: componentBackground,
+      outline: 'none',
+      boxShadow,
+      transition: `background ${motionDurationSlow},width ${motionDurationSlow} cubic-bezier(0.2, 0, 0, 1) 0s`,
```
Comment (zombieJ) [inline components/menu/style/index.tsx:202]: `FIXME`

File: components/menu/style/index.tsx [hunk 6]
```diff
@@ -1,6 +1,1064 @@
...
+      ...clearFix(),
+
+      // sub-menu
+      [`${componentCls}`]: {
+        color: '#000',
```
Comment (zombieJ) [inline components/menu/style/index.tsx:207]: `FIXME`

File: package.json [hunk 7]
```diff
@@ -136,7 +136,7 @@
     "rc-input": "~0.0.1-alpha.5",
     "rc-input-number": "~7.3.0",
     "rc-mentions": "~1.7.0",
-    "rc-menu": "~9.5.1",
+    "rc-menu": "^9.5.4",
```
Comment (MadCcc) [inline package.json:139]:
```suggestion
    "rc-menu": "~9.5.4",
```
这里应该可以不用改

File: components/menu/style/index.tsx [hunk 8]
```diff
@@ -1,6 +1,1070 @@
...
+        [`${componentCls}-item-selected a`]: {
+          color: primaryColor,
+        },
+      },
+
+      '&::after': {
+        display: 'block',
+        clear: 'both',
+        height: 0,
+        content: `' '`, // FIXME  '\20'
```
Comment (MadCcc) [inline components/menu/style/index.tsx:815]: 这个是不是可以写进去？

Comment (heiyu4585) [reply components/menu/style/index.tsx:815]: done

File: package.json (hunk 7 above)
Comment (heiyu4585) [reply package.json]: done

**Review by MadCcc — APPROVED** [review]

**Comment by zombieJ** [comment]
啧啧，有几个例子挂了。我看看去

**Comment by heiyu4585** [comment]
> 啧啧，有几个例子挂了。我看看去

get

**Comment by zombieJ** [comment]
Rebase 一下~~

**Comment by heiyu4585** [comment]
> Rebase 一下~~

ready

**Comment by zombieJ** [comment]
https://preview-34958-ant-design.surge.sh/components/menu-cn/#components-menu-demo-switch-mode  这个例子切换一下样式丢了：
截屏2022-04-19 下午3 40 23

**Comment by zombieJ** [comment]
rebase 一下~

**Comment by heiyu4585** [comment]
> rebase 一下~

ready

**Comment by zombieJ** [comment]
Site 的样式挂了，感觉是优先级问题要调整一下：
截屏2022-04-20 下午3 27 42

**Comment by heiyu4585** [comment]
> > Site 的样式挂了，感觉是优先级问题要调整一下： 截屏2022-04-20 下午3 27 42
> 
> 这个额外 PR 搞好了，这个先合了~

get

**Pull request merged**